File: patch_role.py
import re

with open("app/src/main/java/com/yofidewo/pos/data/Entities.kt", "r") as f:
    content = f.read()

old_role = """data class RoleEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String, // "Admin", "Kasir", etc.
    val canEdit: Boolean = false,
    val canDelete: Boolean = false,
    val canCreate: Boolean = false,
    val canManageProducts: Boolean = false,
    val canViewReports: Boolean = false,
    val canManageStock: Boolean = false,
    val canReturnSales: Boolean = false
)"""

new_role = """data class RoleEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val canViewDashboard: Boolean = false,
    val canViewCashier: Boolean = false,
    val canViewProducts: Boolean = false,
    val canViewReports: Boolean = false,
    val canViewSettings: Boolean = false
)"""

content = content.replace(old_role, new_role)

with open("app/src/main/java/com/yofidewo/pos/data/Entities.kt", "w") as f:
    f.write(content)

File: patch_role.kt
import com.yofidewo.pos.data.RoleEntity

val roleAdminId = roleDao.insertRole(RoleEntity(name = "Administrator", canEdit = true, canDelete = true, canCreate = true, canManageProducts = true, canViewReports = true, canManageStock = true, canReturnSales = true))
val roleKasirId = roleDao.insertRole(RoleEntity(name = "Kasir", canEdit = false, canDelete = false, canCreate = true, canManageProducts = false, canViewReports = false, canManageStock = false, canReturnSales = false))
